synapse/handlers/message.py

```diff
@@ -805,11 +811,15 @@ def handle_new_client_event(
             )
 
         try:
-            yield self.auth.check_from_context(room_version, event, context)
+            yield self.auth.check_from_context(room_version.identifier, event, context)
         except AuthError as err:
             logger.warning("Denying new event %r because %s", event, err)
             raise err
 
+        # Ensure the data is spec compliant JSON.
+        if room_version.strict_canonicaljson:
```

Comment:
I'm unconvinced this is the right place for this check. We do a bunch of other checks on new events in `EventValidator.validate_new`, which is called before we get to this point. Why not add the new check alongside the existing ones?

Reply:
Because I did not find that code. I'll see if I can fit it in there.

Reply:
I added it here and that seems to actually catch a few more locations. Thanks for the pointer!